src/components/typography/typography.style.ts

```diff
@@ -0,0 +1,194 @@
+import styled, { css } from "styled-components";
+import { space } from "styled-system";
+import styledColor from "../../style/utils/color";
+import baseTheme from "../../style/themes/base";
+import { TypographyProps, VariantTypes } from "./typography.component";
+
+const getAs = (variant?: VariantTypes) => {
+  switch (variant) {
+    case "h1-large":
+      return "h1";
+    case "segment-header":
+    case "segment-header-small":
+    case "segment-subheader":
+    case "segment-subheader-alt":
+      return "h5";
+    case "big":
+      return "p";
+    default:
+      return variant;
+  }
+};
+
+const getSize = (variant?: VariantTypes) => {
+  switch (variant) {
+    case "h1-large":
+      return "32px";
+    case "h1":
+      return "24px";
+    case "h2":
+      return "22px";
+    case "h3":
+    case "segment-header":
+      return "20px";
+    case "h4":
+    case "segment-header-small":
+      return "18px";
+    case "h5":
+    case "segment-subheader":
+    case "big":
+      return "16px";
+    case "small":
+    case "sub":
+    case "sup":
+      return "13px";
+    case "segment-subheader-alt":
+    case "p":
+    case "b":
+    case "strong":
+    case "em":
+    default:
+      return "14px";
+  }
+};
+
+const getLineHeight = (variant?: VariantTypes) => {
+  switch (variant) {
+    case "h1-large":
+      return "40px";
+    case "h1":
+    case "segment-subheader":
+      return "31px";
+    case "h2":
+      return "29px";
+    case "h3":
+    case "segment-header":
+      return "26px";
+    case "big":
+      return "24px";
+    case "h4":
+    case "segment-header-small":
+      return "23px";
+    case "small":
+    case "sub":
+    case "sup":
+      return "20px";
+    case "h5":
+    case "segment-subheader-alt":
+    case "p":
+    case "b":
+    case "strong":
+    case "em":
+    default:
+      return "21px";
+  }
+};
+
+const getWeight = (variant?: VariantTypes) => {
+  switch (variant) {
+    case "h1-large":
+    case "h1":
+    case "segment-header":
+    case "segment-header-small":
+      return "900";
+    case "h2":
+    case "h3":
+    case "segment-subheader":
+    case "segment-subheader-alt":
+    case "b":
+    case "em":
+    case "strong":
+      return "700";
+    case "h4":
+    case "h5":
+    case "p":
+    case "small":
+    case "big":
+    case "sub":
+    case "sup":
+    default:
+      return "400";
+  }
+};
+
+const getTransform = (variant?: VariantTypes) => {
+  if (variant === "segment-subheader-alt") {
+    return "uppercase";
+  }
+  return "none";
+};
+
+const getDecoration = (variant?: VariantTypes) => {
+  if (variant === "em") {
+    return "underline";
+  }
+  return "none";
+};
+
+const StyledTypography = styled.span.attrs(
+  ({
+    variant,
+    as,
+    fontSize,
+    fontWeight,
+    textTransform,
+    lineHeight,
+    textDecoration,
+  }: TypographyProps) => {
+    return {
+      as: as || getAs(variant),
```

Comment:
**thought**: perhaps when we come to do FE-6177, we should consider deprecating `as` here anyway since we have the `variant` prop